Add codebyte examples for hoisted variables and functions (#4898)
Codecademy/docs: content/javascript/concepts/hoisting/hoisting.md

var myName = 'Brandon';
```

### Codebyte Example

This example demonstrates how a variable can be declared after it is used. Try changing the placement of the variable declaration (`var name;`) and initialization (`name = 'Brandon';`) to see how it affects the output (hint: move the initialization to after the `console.log` - what does it output? Do you get an error? What about if you remove the declaration entirely?).

```codebyte/javascript
name = 'Brandon'

> 
> console.log(`My name is ${name} and I am logging a hoisted variable!`);
> 
> var name;

Reviewer: ```suggestion
console.log(`My name is ${name} and I am logging a hoisted variable!`);
var name;
```

How is that possible? Again, at this point in the execution phase, the hoisted `greetings` variable is of type `undefined`. The definition of greetings as a function doesn't happen until the _next_ line. Therefore, JavaScript threw the `TypeError`.

### Codebyte Example

> This is a very simple example of how a function declaration can be hoisted. Try changing the function declaration to a function expression to see how that affects the output.

Reviewer: ```suggestion

This is an example of how a function declaration can be hoisted. Try changing the function declaration to a function expression to see how that affects the output.
```

```

Accessing variables and functions before they are declared is bad practice because it is hard to understand and confusing for people who read your code. With `var`, you may read or change the variable unintentionally before initialization, but JavaScript doesn't give you an error that you may be doing something wrong. Thanks to `let`, `const`, and TDZ, we can avoid and catch many potential bugs.

### Codebyte Example

> This is a simple example of hoisting with a `let` variable declaration. Try running it and notice that no error is thrown. What about if you move the declaration to after the `console.log` statement? How does this compare with using `var`?

Reviewer: ```suggestion
This is an example of hoisting with a `let` variable declaration. Try running it and notice that no error is thrown.
```
